src/infer_request.cc

```diff
@@ -420,6 +420,14 @@ InferenceRequest::Run(std::unique_ptr<InferenceRequest>& request)
   return status;
 }
 
+FailureReason stringToFailureReason(const std::string& error_type) {
+  if (error_type == "REJECTED") return FailureReason::REJECTED;
+  if (error_type == "CANCELED") return FailureReason::CANCELED;
+  if (error_type == "BACKEND") return FailureReason::BACKEND;
+  if (error_type == "TIMEOUT") return FailureReason::TIMEOUT;
+  return FailureReason::OTHER;
+}
```

Comment:
I don't have a link to back it up right now, but I think we should generally avoid these c-style 1-line if statements, and instead have the brackets `{}`

Reply:
Corrected.